pkg/apis/openstack/validation/infrastructure_test.go

```diff
@@ -196,15 +197,30 @@ var _ = Describe("InfrastructureConfig validation", func() {
 			}
 			infrastructureConfig.FloatingPoolName = floatingPoolName1
 
-			errorList := ValidateInfrastructureConfigAgainstCloudProfile(infrastructureConfig, differentRegion, cloudProfileConfig, nilPath)
+			errorList := ValidateInfrastructureConfigAgainstCloudProfile(infrastructureConfig, domain, differentRegion, cloudProfileConfig, nilPath)
 
 			Expect(errorList).To(ConsistOfFields(Fields{
 				"Type":  Equal(field.ErrorTypeNotSupported),
 				"Field": Equal("floatingPoolName"),
 			}))
 		})
 
-		It("should forbid using the non-regional floating pool name if region is specified", func() {
+		It("should forbid using floating pool name if no floating pools are specified", func() {
+			cloudProfileConfig.Constraints = api.Constraints{
+				FloatingPools: nil,
+			}
+			infrastructureConfig.FloatingPoolName = floatingPoolName1
+
+			errorList := ValidateInfrastructureConfigAgainstCloudProfile(infrastructureConfig, domain, region, cloudProfileConfig, nilPath)
+
+			Expect(errorList).To(ConsistOfFields(Fields{
+				"Type":  Equal(field.ErrorTypeNotSupported),
+				"Field": Equal("floatingPoolName"),
+			}))
+		})
+
+		It("should always allow using the non-regional floating pool name", func() {
```

Comment:
Hmm, but if the floating pool has a domain then it should only be allowed if the domain matches, or?

Reply:
yes, see test "should only allow domain specific floating pool name for this domain"
I will rename this to "should always allow using a global floating pool name in presence of a regional fp"

Reply:
+1